presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlock.java

```diff
@@ -280,4 +280,26 @@ public String toString()
         sb.append('}');
         return sb.toString();
     }
+
+    @Override
+    public Block getLoadedBlock()
+    {
+        Block loadedKeyBlock = keyBlock.getLoadedBlock();
```

Comment:
Note keyBlock must have loaded since today MapBlock construct hash table eagerly.

Reply:
Add comment and assert